src/org/daisy/dotify/formatter/impl/sheet/SheetDataSource.java

```diff
@@ -420,13 +415,14 @@ private boolean ensureBuffer(int index) {
         return true;
     }
 
-    private void setPreviousSheet(int start, int p, DefaultContext rcontext) {
-        int i = 0;
-        //TODO: simplify this?
-        for (int x = start; i < p && x > 0; x--) {
-            SheetIdentity si = new SheetIdentity(rcontext.getSpace(), rcontext.getCurrentVolume(), volumeGroup, x);
-            rcontext.getRefs().keepBreakable(si, false);
-            i++;
+    private static void keepWithPreviousSheets(List<SheetIdentity> previousSheets, int n, DefaultContext rcontext) {
```

Comment:
I'd prefer a more descriptive name than just `n` so that it's clear what `n` stands for
Or, add Javadoc

Reply:
I'll add a Javadoc.

Reply:
@PaulRambags See https://github.com/bertfrees/dotify.library/commit/bb605b079edf38b6eacb3db172792ef701bb697c.